vmc.sh

```diff
@@ -8,21 +8,38 @@ build_requires:
   - CMake
   - "Xcode:(osx.*)"
   - alibuild-recipe-tools
+prepend_path:
+  ROOT_INCLUDE_PATH: "$VMC_ROOT/include/vmc"
 ---
 #!/bin/bash -e
-cmake "$SOURCEDIR"                             \
-  -DCMAKE_CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
-  -DCMAKE_INSTALL_PREFIX="$INSTALLROOT"        \
-  -DCMAKE_INSTALL_LIBDIR=lib                   \
+
+root_version=$(root-config --version)
+root_version=${root_version:0:1}
+
+if (( ${root_version} == 6 )) ; then
+  cmake "$SOURCEDIR"                             \
+    -DCMAKE_CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+    -DCMAKE_INSTALL_PREFIX="$INSTALLROOT"        \
+    -DCMAKE_INSTALL_LIBDIR=lib                   \
     ${CXXSTD:+-DCMAKE_CXX_STANDARD=$CXXSTD}
 
-cmake --build . -- ${JOBS:+-j$JOBS} install
+  cmake --build . -- ${JOBS:+-j$JOBS} install
+  ln -s $INSTALLROOT/lib/libVMCLibrary.so $INSTALLROOT/lib/libVMC.so
 
-# Modulefile
-MODULEDIR="$INSTALLROOT/etc/modulefiles"
-MODULEFILE="$MODULEDIR/$PKGNAME"
-mkdir -p "$MODULEDIR"
-alibuild-generate-module --bin --lib > $MODULEFILE
-cat >> "$MODULEFILE" <<EoF
-setenv VMC_ROOT \$PKG_ROOT
+  # Modulefile
+  MODULEDIR="$INSTALLROOT/etc/modulefiles"
+  MODULEFILE="$MODULEDIR/$PKGNAME"
+  mkdir -p "$MODULEDIR"
+  alibuild-generate-module --bin --lib > $MODULEFILE
+  cat >> "$MODULEFILE" <<EoF
+  setenv VMC_ROOT \$PKG_ROOT
+  prepend-path ROOT_INCLUDE_PATH \$PKG_ROOT/include/vmc
 EoF
+
+else
```

Comment:
Is this approach acceptable?

Reply:
apart from the fact I would do:

```bash
case $ROOT_VERSION in
  6*) ... ;;
  *) .. ;;
esac
```

it seems fine to me. You could also generate the module file once, before actually building the package.

Reply:
Modulfile is created at the beginning and only updated with the corresponding env variables if VMC is actually built